### PR TITLE
Jira source fixes

### PR DESCRIPTION
**Pull request by Galactus22625**
### Description
Changes to JiraSource
Switch some logs to trace
Removed JiraProjectCache and stop bringing projects into pipeline since some projects had a null pointer issue
fix issue that occurred when accountUrl didn't end with a slash
 
### Issues Resolved
Resolves #[Issue number to be closed when this PR is merged]
 
### Check List
- [x] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by san81 — APPROVED** [review]
Thank you for taking care of these fixes

File: data-prepper-plugins/saas-source-plugins/jira-source/src/main/java/org/opensearch/dataprepper/plugins/source/jira/JiraIterator.java [hunk 1]
```diff
@@ -47,7 +48,11 @@ public boolean hasNext() {
 
     @Override
     public ItemInfo next() {
-        return this.itemInfoQueue.remove();
+        if (hasNext()) {
+            return this.itemInfoQueue.remove();
```
[the remark this answers is not in the document]
Comment (Galactus22625) [reply data-prepper-plugins/saas-source-plugins/jira-source/src/main/java/org/opensearch/dataprepper/plugins/source/jira/JiraIterator.java:52]: yes, it is tested in testItemInfoQueueNotEmpty in JiraIteratorTest.

[the remark this answers is not in the document]
Comment (Galactus22625) [reply data-prepper-plugins/saas-source-plugins/jira-source/src/main/java/org/opensearch/dataprepper/plugins/source/jira/JiraIterator.java:52]: should I break this test into two tests

[the remark this answers is not in the document]
Comment (Galactus22625) [reply data-prepper-plugins/saas-source-plugins/jira-source/src/main/java/org/opensearch/dataprepper/plugins/source/jira/JiraIterator.java:52]: added tests

**Review by sb2k16 — APPROVED** [review]

**Comment by Galactus22625** [comment]
merge after pent test over

**Comment by Galactus22625** [comment]
> merge after pent test over

over today after 5

**Review by kkondaka — APPROVED** [review]

**Pull request merged**